runtime/ts/type.ts

```diff
@@ -64,7 +64,8 @@ export class Type {
     }
     if (tag === 'Variable') {
       if (!(data instanceof TypeVariable)) {
-        data = new TypeVariable(data.name, data.constraint);
+        // TODO determine if 3rd param should be null?
```

Comment:
any advice for this?

Reply:
@shans should be able to help.

Reply:
Okay, added as a reviewer.

Reply:
Yeah, I think that the 3rd param should be null here; but it's not immediately clear why, so would you mind adding a comment at the same time:
```
// type constraints ("~a with EntityName") should be considered minimum requirements
// for the type, so are fed in as 'canWriteSuperset' (i.e. low-watermark) constraints.
```

That's still a bit cryptic sorry :/ but at least it provides a hint of what's going on.

Reply:
thanks, done.